src/test/java/step2/StringAddCalculatorTest.java

```diff
@@ -0,0 +1,72 @@
+package step2;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class StringAddCalculatorTest {
+
+    private Integer getInt(String value) throws NumberFormatException {
+        return Integer.parseInt(value);
+    }
+
+    @Test
+    @DisplayName("Return 0 When input value is Null or Empty")
+    void nullOrEmpty() {
+        Integer expected = 0;
+        assertThat(StringAddCalculator.splitAndSum(null)).isEqualTo(expected);
+        assertThat(StringAddCalculator.splitAndSum("")).isEqualTo(expected);
+    }
+
+    @Test
+    @DisplayName("Return the Number When input value is just a number")
+    void oneNumber() throws NumberFormatException {
+        String number = "3";
+        assertThat(StringAddCalculator.splitAndSum(number)).isEqualTo(getInt(number));
+    }
+
+    @Test
+    @DisplayName("Return Sum of Numbers When input value is concatenated numbers by comma")
+    void concatenatedByComma() {
+        String num1 = "1";
+        String num2 = "5";
+        String delimiter = ",";
+        assertThat(StringAddCalculator.splitAndSum(num1 + delimiter + num2))
+                .isEqualTo(getInt(num1) + getInt(num2));
+    }
+
+    @Test
+    @DisplayName("Return Sum of Numbers When input value is concatenated numbers by comma or colon")
+    void concatenatedByCommaOrColon() {
+        String num1 = "1";
+        String num2 = "3";
+        String num3 = "7";
+        String delimiter1 = ",";
+        String delimiter2 = ":";
+        assertThat(StringAddCalculator.splitAndSum(num1 + delimiter1 + num2 + delimiter2 + num3))
+                .isEqualTo(getInt(num1) + getInt(num2) + getInt(num3));
+    }
+
+    @Test
+    @DisplayName("Return Sum of Numbers When input value is concatenated numbers by custom delimiter")
+    void concatenatedByCustomDelimiter() {
+        String num1 = "3";
+        String num2 = "8";
+        String customDelimiter = "-";
+        String customDelimiterFormat = "//%s\n";
+        assertThat(StringAddCalculator.splitAndSum(String.format(customDelimiterFormat, customDelimiter) + num1 + customDelimiter + num2))
```

Comment:
```suggestion
        assertThat(StringAddCalculator.splitAndSum(String.format(customDelimiterFormat, customDelimiter) + num1 + customDelimiter + num2))
```

테스트 코드에 비지니스 로직이 있네요.
이러한 부분은 문제가 될 수 있어요.

프로덕션 코드와 테스트 코드에서 사용하는 `포맷("//%s\n")`이 서로 달라질수 있기 때문이에요.

그리고, 항상 프로덕션코드와 테스트코드를 같이 신경써야하는 부분도 TDD를 지속하기에 힘든 부분이 될수 있다고 생각해요.

테스트 하고자하는 메소드에 집중할수있도록,  `input, output` 값을 단순하게 유지하는 고민을 해보면 어떨까요?!

Reply:
@wooobo 
프로덕션 코드와 테스트 코드에서 사용하는 포맷이 달라질 수 있는 부분은 어떻게 관리하는 게 좋을까요?
특정 포맷이 입력됐을 때, 특정 비즈니스 로직을 수행하게 하는 것을 테스트 하려면
테스트 코드에서도 해당 포맷이 정의되어 있어야 테스트가 가능할 것 같아서요.